Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Validator.rst

```diff
@@ -67,12 +71,28 @@ The validator is used as annotation in the action methods of the controller:
 ..  literalinclude:: _CustomValidator/_ObjectValidatorUsage.php
     :caption: EXT:blog_example/Classes/Controller/BlogController.php, modified
 
+..  _extbase_domain_validator-di:
+
 Dependency injection in validators
 ==================================
 
-Starting with TYPO3 v12 Extbase validators are capable of :ref:`dependency injection <Dependency-Injection>`
+Extbase validators are capable of :ref:`dependency injection <Dependency-Injection>`
 without further configuration, you can use the constructor method:
 
 ..  literalinclude:: _Validator/_MyCustomValidator.php
-    :language: php
+    :caption: EXT:my_extension/Classes/Validators/MyCustomValidator.php
```

Comment:
To be compatible with your saying somewhere above I prefer adding "Domain" here, too.

```suggestion
    :caption: EXT:my_extension/Classes/Domain/Validators/MyCustomValidator.php
```